boards/silabs/radio_boards/siwx917_rb4338a/siwx917_rb4338a.dts

```diff
@@ -97,3 +97,22 @@
 		};
 	};
 };
+
+&flash0 {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		// NWP (Network Processor) partition (do not overwrite)
+		nwp_firmware_partition: partition@0 {
```

Comment:
Is this correct? The NWP sees the flash at a different address (in the 0x4xxx range), where it has a 1.8 MB image that the M4 can't access. As I understand it, the memory map as seen by the M4 actually does start at 0x08202000, not 0x08200000.

Reply:
I have not found proper reference for that.

If I understand right, the "917 reference manual" says the flash start at 0x08000000 (4.3.4 Address Mapping). The size NWP firmware (in fact the size of the .rps file) is 1.6MB. So, it would probably makes sense to reserve 2MB and place the M4 firmware at 0x08200000. However, isp_prepare.py prefix the firmware with 4096 bytes for the rps header (I believe it could be decreased). So, the firmware should be placed at 0x08201000.

BTW, the datasheet says siwg917m111mgtba has 8MB of flash.

Reply:
The important thing to realize is that in common flash mode, the NWP owns the flash. The M4 is a subordinate that is only allowed to use the subset the NWP gives it access to.